### PR TITLE
Implementation of flip function 

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR adds `flip` functions according to [Python array API standard](https://data-apis.org/array-api/latest/API_specification/generated/signatures.manipulation_functions.flip.html) for usm_ndarray.

### CODE REVIEW

File: dpctl/tensor/_manipulation_functions.py [hunk 1]
```diff
@@ -207,3 +207,25 @@ def broadcast_arrays(*args):
...
+    returns a view of X with the entries of axis reversed.
+    """
+    if not isinstance(X, dpt.usm_ndarray):
+        raise TypeError(f"Expected usm_ndarray type, got {type(X)}.")
+    X_ndim = X.ndim
+    if axes is None:
+        indexer = (np.s_[::-1],) * X_ndim
+    else:
+        axes = normalize_axis_tuple(axes, X_ndim)
+        indexer = [np.s_[:]] * X.ndim
```
Comment (oleksandr-pavlyk) [inline dpctl/tensor/_manipulation_functions.py:227]: More direct `indexer = tuple( np.s_[::-1] if i in axes else np.s_[:] for i in range(X.ndim))`.

This way a list temporary is not needed.

Comment (vlad-perevezentsev) [reply dpctl/tensor/_manipulation_functions.py:227]: Great remark. Thanks

**Comment by vlad-perevezentsev** [comment]
Tests were added for an empty array with 0-size dimensions
`test_flip_0d` was updated

**Review by oleksandr-pavlyk — APPROVED** [review]
LGTM! Thanks @vlad-perevezentsev

**Pull request merged**